lib/layer2.js

```diff
@@ -1,36 +1,99 @@
 const configManager = require('./config-manager')
 const ph = require('./plugin-helper')
 const _ = require('lodash/fp')
-const logger = require('./logger')
 
-function fetch (settings, cryptoCode) {
+class Layer2Composite {
```

Comment:
We're not using JavaScript classes anywhere in the code, mostly because we don't like the object-oriented paradigm, so let's not add classes in.

I think this is getting too complicated with layer2-composite and wallet-adapter.js. It seems like there a lot of logic in wallet.js and we also don't want to be in a situation where layer2 depends on wallet.js and wallet.js also depends on layer2. Therefore, it seems like the easiest solution is to leave everything in wallet.js, as it was before.